### PR TITLE
Issue #371 - Documentation for alternate configuration directory

### PR DESCRIPTION
**Pull request by rakillen**

### CODE REVIEW

File: site/config/custom_configuration.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+## Custom Configuration
+
+WDT allows you to create or extend the pre-installed type definitions, model filters, variable injectors, and target environments. Starting with WDT 1.10.0, these additional configuration files can be stored outside the `$WLSDEPLOY_HOME/lib` directory. This allows the files to remain in place if the WDT installation is moved or upgraded to a new version.
+
+To use a separate configuration directory, set the `WDT_CUSTOM_CONFIG` environment variable to the directory to be used for configuration. For example:
+```
+export WDT_CUSTOM_CONFIG=/etc/wdtconfig
+```
+
+The customized configuration files should be named and organized the same way they would be under the `$WLSDEPLOY_HOME/lib` directory. For example:
```
Comment (ddsharpe) [inline site/config/custom_configuration.md:10]: We should make it more obvious that the file name is how the CLA selects it by name.

File: site/config/custom_configuration.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    model_variable_injector.json
+    variable_keywords.json
+```
+This is a full set of files that can be configured, you will only need to add the files you have created or extended.
+
+The WDT tools will look for each configuration file under `$WDT_CUSTOM_CONFIG` if specified, then under `$WLSDEPLOY_HOME/lib`.
+
+### Example: Extending a Type Definition
+
+If you want to extend the `WLS` type definition to add more extension templates, you would first make a copy of `$WLSDEPLOY_HOME/lib/typedefs/WLS.json`, and make your edits to the new file. Copy the new file to `$WDT_CUSTOM_CONFIG/typedefs/MY_WLS.json`. Set the `$WDT_CUSTOM_CONFIG` environment variable, and run the tool referencing the new type definition:
```
Comment (ddsharpe) [inline site/config/custom_configuration.md:31]: The ordering of the steps might be a little easier to follow if it were:
Create WDT_CUSTOM_CONFIG
Copy from lib to custom, renaming if desired
Edit content of new file

File: site/config/custom_configuration.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        *.json  (typedef files)
+    targets
+        my-target
+            target.json
+            *.py  (filter files)
+    model_filters.json
+    model_variable_injector.json
+    variable_keywords.json
+```
+This is a full set of files that can be configured, you will only need to add the files you have created or extended.
```
Comment (rosemarymarano) [inline site/config/custom_configuration.md:25]: configured, you -> configured; you (these are two sentences; if you want them joined, use a semi-colon.)
only need -> need only

File: site/prepare.md [hunk 4]
```diff
@@ -32,6 +32,8 @@ You can then customize the `k8s_variable.properties` and `create_k8s_secrets.sh`
 
 The '-target' parameter is referring to a file on the file system '$WLSDEPLOY_HOME/lib/target/<target value>/target.json'
 
+A target may also be configured in a [Custom Configuration](config/custom_configuration.md) directory, in `$WDT_CUSTOM_CONFIG/target/<target name>/target.json`.
```
Comment (rosemarymarano) [inline site/prepare.md:35]: A target may also be configured in a -> You also can configure a target in a

File: site/prepare.md [hunk 5]
```diff
@@ -32,6 +32,8 @@ You can then customize the `k8s_variable.properties` and `create_k8s_secrets.sh`
 
 The '-target' parameter is referring to a file on the file system '$WLSDEPLOY_HOME/lib/target/<target value>/target.json'
 
+A target may also be configured in a [Custom Configuration](config/custom_configuration.md) directory, in `$WDT_CUSTOM_CONFIG/target/<target name>/target.json`. 
+
 It has the format
```
Comment (rosemarymarano) [inline site/prepare.md:37]: format -> format:

File: site/prepare.md [hunk 6]
```diff
@@ -32,6 +32,8 @@ You can then customize the `k8s_variable.properties` and `create_k8s_secrets.sh`
 
 The '-target' parameter is referring to a file on the file system '$WLSDEPLOY_HOME/lib/target/<target value>/target.json'
 
+A target may also be configured in a [Custom Configuration](config/custom_configuration.md) directory, in `$WDT_CUSTOM_CONFIG/target/<target name>/target.json`. 
+
```
Comment (rosemarymarano) [inline site/prepare.md:36]: Two other items of note to be corrected in prepare.md:
json -> JSON (please correct this globally)
In the output directory, you will find -> In the output directory, you will find: (stem sentences required a trailing colon.)

File: site/config/custom_configuration.md (hunk 1 above)
Comment (rakillen) [reply site/config/custom_configuration.md:10]: Several of these are not specified in CLA. I broke out bullets for each type, and referenced other docs for details. I linked the typedef example here, that goes into more detail about CLS name.

File: site/config/custom_configuration.md (hunk 2 above)
Comment (rakillen) [reply site/config/custom_configuration.md]: Broke these out into bullets as described.

File: site/config/custom_configuration.md (hunk 3 above)
Comment (rakillen) [reply site/config/custom_configuration.md]: two sentences is better thx

**Pull request merged**